frame/ddc-staking/src/benchmarking.rs

```diff
@@ -0,0 +1,218 @@
+	&'static str,
+> {
+	// Clean up any existing state.
+	clear_storages_and_edges::<T>();
+
+	// Create new storages
+	let mut storages: Vec<<T::Lookup as StaticLookup>::Source> =
+		Vec::with_capacity(n_storages as usize);
+	for i in 0..n_storages {
+		let (stash, controller) = create_stash_controller::<T>(i + SEED, 100)?;
+		let storage_prefs = StoragePrefs { foo: true };
+		DddcStaking::<T>::store(RawOrigin::Signed(controller).into(), storage_prefs)?;
+		let stash_lookup: <T::Lookup as StaticLookup>::Source = T::Lookup::unlookup(stash);
+		storages.push(stash_lookup);
+	}
+
+	// Create new edges
+	let mut edges: Vec<<T::Lookup as StaticLookup>::Source> = Vec::with_capacity(n_edges as usize);
+	for i in 0..n_edges {
+		let (stash, controller) = create_stash_controller::<T>(i + SEED, 100)?;
+		let edge_prefs = EdgePrefs { foo: true };
+		DddcStaking::<T>::serve(RawOrigin::Signed(controller).into(), edge_prefs)?;
+		let stash_lookup: <T::Lookup as StaticLookup>::Source = T::Lookup::unlookup(stash);
+		edges.push(stash_lookup);
+	}
+
+	Ok((storages, edges))
+}
+
+struct ListScenario<T: Config> {
+	/// Stash that is expected to be moved.
+	origin_stash1: T::AccountId,
+	/// Controller of the Stash that is expected to be moved.
+	origin_controller1: T::AccountId,
+	dest_weight: BalanceOf<T>,
+}
+
+impl<T: Config> ListScenario<T> {
+	fn new(origin_weight: BalanceOf<T>) -> Result<Self, &'static str> {
+		ensure!(!origin_weight.is_zero(), "origin weight must be greater than 0");
+
+		// burn the entire issuance.
+		let i = T::Currency::burn(T::Currency::total_issuance());
+		sp_std::mem::forget(i);
+
+		// create accounts with the origin weight
+
+		let (origin_stash1, origin_controller1) =
+			create_stash_controller_with_balance::<T>(USER_SEED + 2, origin_weight)?;
+
+		let (_origin_stash2, _origin_controller2) =
+			create_stash_controller_with_balance::<T>(USER_SEED + 3, origin_weight)?;
+
+		// create an account with the worst case destination weight
+		let (_dest_stash1, _dest_controller1) =
+			create_stash_controller_with_balance::<T>(USER_SEED + 1, origin_weight)?;
+
+		Ok(ListScenario { origin_stash1, origin_controller1, dest_weight: origin_weight })
+	}
+}
+
+const USER_SEED: u32 = 999666;
+
+benchmarks! {
+	bond {
+		let stash = create_funded_user::<T>("stash", USER_SEED, 100);
+		let controller = create_funded_user::<T>("controller", USER_SEED, 100);
+		let controller_lookup: <T::Lookup as StaticLookup>::Source
+			= T::Lookup::unlookup(controller.clone());
+		let amount = T::Currency::minimum_balance() * 10u32.into();
+		whitelist_account!(stash);
+	}: _(RawOrigin::Signed(stash.clone()), controller_lookup, amount)
+	verify {
+		assert!(Bonded::<T>::contains_key(stash));
+		assert!(Ledger::<T>::contains_key(controller));
+	}
+
+	bond_extra {
+		// clean up any existing state.
+		clear_storages_and_edges::<T>();
+
+		let origin_weight = MinStorageBond::<T>::get().max(T::Currency::minimum_balance());
```

Comment:
What is original weight?

Reply:
I will rename as it's not related to our codebase